app/controllers/categories_controller.rb

```diff
@@ -1,13 +1,13 @@
 # frozen_string_literal: true
 
 class CategoriesController < ApplicationController
-	def show
-		@child_categories = find_child_categories(cat_id=params[:id])
-	end
+  def show
+    @child_categories = find_child_categories(cat_id = params[:id])
```

Comment:
Tutaj żeby było bardziej obiektowo zrobiłbym
```
@category = Category.find(params[:id])
```
a w widoku korzystał z `@category.child_categories`

żeby metoda #child_categories była dostępna należy dodać:

```ruby
class Category < ApplicationRecord
  has_many :child_categories, class_name: 'Category', foreign_key: 'parent'
end
```

Reply:
Szybka podpowiedź jak wyczuć, że programujemy strukturalnie a nie obiektowo: 
- do funkcji przekazujemy proste typy jak integer zamiast całych obiektów (jak `find_child_categories`, które przyjmuje integera)
- funkcje wywołujemy w abstrakcyjnych klasach a nie obiektach, które są z nimi mocno powiązane (czyli `find_child_categories` jest funkcją zdefiniowaną w kontrolerze, który nie jest żadnym z naszych modeli, a ta fukcja jest ściśle powiązana z modelem `Category`)

Gdyby to nie były railsy to można by zrobić metodę w klasie Category:
```ruby
class Category < ApplicationRecord
  def child_categories
    Category.where(parent: self.id) # self nie jest potrzebne, samo Category.where(parent: id) też by działało
  end
end
```

ale skoro korzystamy z Rails to możemy użyć wbudowanego helpera do asocjacji - `has_many`

Reply:
I ostatnia uwaga:
jeśli jednak chcieliby Państwo definiować funkcje w kontrolerze to inaczej przekazuje się parametry do funkcji. Skoro funkcja jest zdefiniowana tak:
```ruby
def find_child_categories(cat_id)
  Category.where(parent: cat_id)
end
```

to jej wywołanie będzie wyglądać tak:
```ruby
 find_child_categories(params[:id])
```

a nie tak:
```ruby
 find_child_categories(cat_id = params[:id])
```